src/engine.cpp

```diff
@@ -1023,7 +1023,9 @@ CommandCost CmdSetVehicleVisibility(DoCommandFlag flags, EngineID engine_id, boo
 	if (!IsEngineBuildable(e->index, e->type, _current_company)) return CMD_ERROR;
 
 	if ((flags & DC_EXEC) != 0) {
-		AssignBit(e->company_hidden, _current_company, hide);
+		if (e->company_hidden.Test(_current_company) != hide) {
+			e->company_hidden.Flip(_current_company);
+		}
```

Comment:
Might be worth implementing an equivalent of std::bitset's `set(x, y)` method for this? I found I needed it in other places...  (Here's my EnumBitSet version...)

```c++
/**
 * Assign the enum value.
 * @param value Enum value to assign.
 * @param set true if the value should be set, false if the value should be reset.
 * @returns The EnumBitset
 */
inline constexpr EnumBitSet &Set(Tenum value, bool set)
{
	return set ? this->Set(value) : this->Reset(value);
}
```